src/js/other-websites/inpage_menu.js

```diff
@@ -51,226 +52,610 @@ async function isUserSignedIn() {
-    sendInPageEvent("click", "input-menu-manage-all-aliases-btn");
-    iframeCloseRelayInPageMenu();
-  });
+      const signedInContentFree = document.querySelector(
+        ".fx-content-signed-in-free"
+      );
 
-  // Create "Get unlimited aliases" link
-  getUnlimitedAliasesBtn.href = `${relaySiteOrigin}/premium?utm_source=fx-relay-addon&utm_medium=input-menu&utm_content=get-premium-link`;
-  getUnlimitedAliasesBtn.addEventListener("click", () => {
-    sendInPageEvent("click", "input-menu-get-premium-btn");
-    iframeCloseRelayInPageMenu();
-  });
+      const signedInContentPremium = document.querySelector(
+        ".fx-content-signed-in-premium"
+      );
+
+      // Create "You have .../.. remaining relay address" message
+      const remainingAliasesSpan = document.querySelector(
+        ".fx-relay-menu-remaining-aliases"
+      );
+
+      const { maxNumAliases } = await browser.storage.local.get(
+        "maxNumAliases"
+      );
+
+      const maskLists = document.querySelectorAll(".fx-relay-menu-masks-list");
+      const masks = await getMasks();
+
+      if (masks.length !== 0) {
+        //
+      }
+
+      maskLists?.forEach(async (maskList) => {
+        // Set Mask List label names
+        const label = maskList.querySelector(".fx-relay-menu-masks-list-label");
+        const stringId = label.dataset.stringId;
+        label.textContent = browser.i18n.getMessage(stringId);
+
+        // Populate mask lists
+        await populateMaskList(maskList, masks);
+      });
 
-  if (!premium) {
-    if (maxNumAliasesReached) {
-      generateAliasBtn.remove();
-      sendInPageEvent("viewed-menu", "input-menu-max-aliases-message");
+      const numAliasesRemaining = maxNumAliases - masks.length;
+      const maxNumAliasesReached = numAliasesRemaining <= 0;
+
+      // Set Generate Mask button
+      buildContent.components.setUnlimitedButton();
+
+      // Free user: Set text informing them how many aliases they can create
       remainingAliasesSpan.textContent = browser.i18n.getMessage(
-        "pageFillRelayAddressLimit_mask",
+        "popupRemainingFreeMasks",
         [numAliasesRemaining, maxNumAliases]
       );
-      // Focus on "Get unlimited alias" button
-      getUnlimitedAliasesBtn.focus();
-    } else {
+
+      const generateAliasBtn = document.querySelector(
+        ".fx-relay-menu-generate-alias-btn"
+      );
+
+      if (maxNumAliasesReached) {
+        generateAliasBtn.remove();
+        sendInPageEvent("viewed-menu", "input-menu-max-aliases-message");
+        remainingAliasesSpan.textContent = browser.i18n.getMessage(
+          "pageNoMasksRemaining"
+        );
+
+        getUnlimitedAliasesBtn.classList.remove("t-secondary");
+        getUnlimitedAliasesBtn.classList.add("t-primary");
+        // Focus on "Get unlimited alias" button
+        getUnlimitedAliasesBtn.focus();
+
+        document.querySelector(".fx-relay-menu-masks-lists").style.order = "2";
```

Comment:
Woah, `order`? What is this sorcery!

Reply:
Heh, though without jokes, reading up on the property I don't understand why it needs to be set here - maybe you could add a comment?

Reply:
(Comment to be added, but explaining here) 

In the UX, there are three+ elements: "Generate more" button, the list(s) and the upgrade button. When the user has maxed out their available masks, the "Generate more" button is hidden/replaced with the upgrade button. The order change is necessary to match the UX. 

<img width="242" alt="image" src="https://user-images.githubusercontent.com/2692333/165351765-637f3101-33b0-473d-9eb0-b7b7ba70d8c1.png">

<img width="243" alt="image" src="https://user-images.githubusercontent.com/2692333/165351772-64d934eb-a7c3-46a6-b7d1-9351c43f9815.png">

Reply:
Comment added in f0903529cedee84c0b0fa0acdb1e291af9ad2dac

Reply:
Let's submit a proposal to TC39 to allow screenshots in code comments! :P